File: sdk/php/src/Resources/Domains.php
<?php

namespace HiTechCloud\SDK\Resources;

class Domains extends BaseResource
{
    /**
     * WHOIS lookup
     */
    public function whois(string $domain): array
    {
        return $this->http->get('/api/whois/' . $domain);
    }

    /**
     * List domains
     */
    public function list(array $params = []): array
    {
        return $this->http->get('/api/domain', $params);
    }

    /**
     * Get domain
     */
    public function get(int $domainId): array
    {
        return $this->http->get('/api/domain/' . $serviceId);
    }

    /**
     * Get domain by name
     */
    public function getByName(string $name): array
    {
        return $this->http->get('/api/domain/name/' . $name);
    }

    /**
     * Get nameservers
     */
    public function getNameservers(int $domainId): array
    {
        return $this->http->get('/api/domain/' . $serviceId . '/ns');
    }

    /**
     * Set nameservers
     */
    public function setNameservers(int $domainId, array $data): array
    {
        return $this->http->put('/api/domain/' . $serviceId . '/ns', $data);
    }

    /**
     * Get EPP code
     */
    public function getEppCode(int $domainId): array
    {
        return $this->http->get('/api/domain/' . $serviceId . '/epp');
    }

    /**
     * Sync domain
     */
    public function sync(int $domainId): array
    {
        return $this->http->get('/api/domain/' . $serviceId . '/sync');
    }

    /**
     * Get registrar lock
     */
    public function getRegLock(int $domainId): array
    {
        return $this->http->get('/api/domain/' . $serviceId . '/reglock');
    }

    /**
     * Set registrar lock
     */
    public function setRegLock(int $domainId, array $data): array
    {
        return $this->http->put('/api/domain/' . $serviceId . '/reglock', $data);
    }

    /**
     * Toggle ID protection
     */
    public function toggleIdProtection(int $domainId, array $data): array
    {
        return $this->http->put('/api/domain/' . $serviceId . '/idprotection', $data);
    }

    /**
     * Get domain contacts
     */
    public function getContacts(int $domainId): array
    {
        return $this->http->get('/api/domain/' . $serviceId . '/contact');
    }

    /**
     * Set domain contacts
     */
    public function setContacts(int $domainId, array $data): array
    {
        return $this->http->put('/api/domain/' . $serviceId . '/contact', $data);
    }

    /**
     * Get email forwarding
     */
    public function getEmailForwarding(int $domainId): array
    {
        return $this->http->get('/api/domain/' . $serviceId . '/email/forwarding');
    }

    /**
     * Set email forwarding
     */
    public function setEmailForwarding(int $domainId, array $data): array
    {
        return $this->http->put('/api/domain/' . $serviceId . '/email/forwarding', $data);
    }

    /**
     * Check availability
     */
    public function checkAvailability(string $domain): array
    {
        return $this->http->get('/api/domain/availability/' . $domain);
    }

    /**
     * List domain orders
     */
    public function listOrders(array $params = []): array
    {
        return $this->http->get('/api/domain/order', $params);
    }

    /**
     * Create domain order
     */
    public function createOrder(array $data): array
    {
        return $this->http->post('/api/domain/order', $data);
    }

    /**
     * Get order form
     */
    public function getOrderForm(int $orderId): array
    {
        return $this->http->get('/api/domain/order/' . $serviceId . '/form');
    }

    /**
     * Renew domain
     */
    public function renew(int $domainId, array $data = []): array
    {
        return $this->http->post('/api/domain/' . $serviceId . '/renew', $data);
    }

    /**
     * WHOIS lookup v2
     */
    public function whoisLookup(string $domain): array
    {
        return $this->http->get('/api/whoislookup/' . $domain);
    }

    /**
     * Get domain documents
     */
    public function getDocuments(int $domainId): array
    {
        return $this->http->get('/api/domain/' . $serviceId . '/documents');
    }

}

File: sdk/java/src/main/java/vn/hitechcloud/sdk/resources/Domains.java
package vn.hitechcloud.sdk.resources;

import java.util.Map;
import vn.hitechcloud.sdk.HttpClient;

public class Domains extends BaseResource {

    public Domains(HttpClient http) {
        super(http);
    }

    /** WHOIS lookup */
    public Map<String, Object> whois(String domain) throws Exception {
        return http.get(String.format("/api/whois/%domain", domain));
    }

    /** List domains */
    public Map<String, Object> domains() throws Exception {
        return http.get("/api/domain");
    }

    /** Get domain */
    public Map<String, Object> domain(int id) throws Exception {
        return http.get(String.format("/api/domain/%id", id));
    }

    /** Get domain by name */
    public Map<String, Object> domainByName(String name) throws Exception {
        return http.get(String.format("/api/domain/name/%name", name));
    }

    /** Get nameservers */
    public Map<String, Object> nameservers(int id) throws Exception {
        return http.get(String.format("/api/domain/%id/ns", id));
    }

    /** Update nameservers */
    public Map<String, Object> updateNameservers(int id, Map<String, Object> data) throws Exception {
        return http.put(String.format("/api/domain/%id/ns", id), data);
    }

    /** Get EPP code */
    public Map<String, Object> eppCode(int id) throws Exception {
        return http.get(String.format("/api/domain/%id/epp", id));
    }

    /** Sync domain */
    public Map<String, Object> syncDomain(int id) throws Exception {
        return http.get(String.format("/api/domain/%id/sync", id));
    }

    /** Get reglock */
    public Map<String, Object> reglock(int id) throws Exception {
        return http.get(String.format("/api/domain/%id/reglock", id));
    }

    /** Update reglock */
    public Map<String, Object> updateReglock(int id, Map<String, Object> data) throws Exception {
        return http.put(String.format("/api/domain/%id/reglock", id), data);
    }

    /** Update ID protection */
    public Map<String, Object> idProtection(int id, Map<String, Object> data) throws Exception {
        return http.put(String.format("/api/domain/%id/idprotection", id), data);
    }

    /** Get domain contacts */
    public Map<String, Object> domainContacts(int id) throws Exception {
        return http.get(String.format("/api/domain/%id/contact", id));
    }

    /** Update domain contacts */
    public Map<String, Object> updateContacts(int id, Map<String, Object> data) throws Exception {
        return http.put(String.format("/api/domain/%id/contact", id), data);
    }

    /** Get email forwarding */
    public Map<String, Object> emailForwarding(int id) throws Exception {
        return http.get(String.format("/api/domain/%id/email/forwarding", id));
    }

    /** Update email forwarding */
    public Map<String, Object> updateEmailForwarding(int id, Map<String, Object> data) throws Exception {
        return http.put(String.format("/api/domain/%id/email/forwarding", id), data);
    }

    /** Check availability */
    public Map<String, Object> availability(String domain) throws Exception {
        return http.get(String.format("/api/domain/availability/%domain", domain));
    }

    /** List domain orders */
    public Map<String, Object> orders() throws Exception {
        return http.get("/api/domain/order");
    }

    /** Create domain order */
    public Map<String, Object> createOrder(Map<String, Object> data) throws Exception {
        return http.post("/api/domain/order", data);
    }

    /** Get order form */
    public Map<String, Object> orderForm(int id) throws Exception {
        return http.get(String.format("/api/domain/order/%id/form", id));
    }

    /** Renew domain */
    public Map<String, Object> renewDomain(int id, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/domain/%id/renew", id), data);
    }

    /** WHOIS lookup v2 */
    public Map<String, Object> whoisLookup(String domain) throws Exception {
        return http.get(String.format("/api/whoislookup/%domain", domain));
    }

    /** Get domain documents */
    public Map<String, Object> documents(int id) throws Exception {
        return http.get(String.format("/api/domain/%id/documents", id));
    }

}
